src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -355,7 +359,11 @@ public Matrix Transform
                     return;
 
                 _currentTransform = value;
-                Canvas.SetMatrix(value.ToSKMatrix());
+
+                // TODO this line is a workaround for bug in DrawingContext.cs
+                _currentTransform *= _dpiScaleMatrix;
```

Comment:
...which is why I suspect you need this here. This is actually not doing the right thing: it's applying the scaling *after* the current transform, whereas it needs to be applied before.